src/services/UserService.ts

```diff
@@ -116,8 +116,9 @@ const updateUser = async (user: User, options: UpdateOptions = {}): Promise<Resp
   }
 
   const response: Response = await httpCurrentUser.put({ entity });
-  if (response.requestSucceeded && typeof user.cookiesConsented === 'boolean') {
-    await PreferencesService.updateUserCookieConsentPreferences({ cookiesConsented: user.cookiesConsented });
+  if (response.requestSucceeded && typeof user.cookiesConsented === 'string') {
```

Comment:
When would this be set to a string? It looks like you're dealing with bools in the places where you can interact with it

Reply:
I discovered the value is cast to a string by the `RadioGroup` component and this wasn't working as expected.

Reply:
Oh I see, that's a bummer. Is there any way you can rectify that within the component that has the radio group? The `UserService` shouldn't be doing strange things because of the behavior of something that is implementing it. Ideally the strange things, like casting the radio group value to a boolean, are done where the effect occurs in the first place.

Reply:
The `RadioGroup` component is part of the MUI library and this comment is included for the `value` prop in the `RadioGroupProps` type:

> Value of the selected radio button. The DOM API casts this to a string.

Reply:
Good point about handling the casting in a more appropriate location, I reverted the change to the `UserService` and handled the issue in the `onChange` handler of the `RadioGroup` component.